File: mobile/src/main/java/com/tilly/steven/stlbusarrivals/Model/Path.java
package com.tilly.steven.stlbusarrivals.Model;

import com.google.android.gms.wearable.DataMap;

import java.util.ArrayList;

/**
 * Created by Steven on 2016-02-04.
 */
public class Path extends ArrayList<Point> {

    public DataMap putData(){
        DataMap map = new DataMap();
        for(int i = 0; i<this.size(); i++){
            map.putDataMap("point"+i,this.get(i).putData());
        }
        return map;
    }
}

File: mobile/src/main/java/com/tilly/steven/stlbusarrivals/model/Path.kt
package com.tilly.steven.stlbusarrivals.model

import com.google.android.gms.wearable.DataMap
import java.util.*

/**
 * Created by Steven on 2016-02-04.
 */
class Path : ArrayList<Point>() {

    fun putData(): DataMap {
        val map = DataMap()
        for (i in 0 until this.size) {
            map.putDataMap("point$i", this[i].putData())
        }
        return map
    }
}
